spring-web/src/main/java/org/springframework/http/RequestEntity.java

```diff
@@ -241,6 +285,17 @@ public static BodyBuilder method(HttpMethod method, URI url) {
 		return new DefaultBodyBuilder(method, url);
 	}
 
+	/**
+	 * Create a builder with the given method and given string base uri template.
+	 * @param method the HTTP method (GET, POST, etc)
+	 * @param uri the uri
+	 * @return the created builder
+	 */
+	public static BodyBuilder method(HttpMethod method, String uri) {
+		return new DefaultBodyBuilder(method, new UriTemplate(uri));
```

Comment:
But why go to `UriTemplate` here, only to go back to `String` in the end in `getUrl()`? There is no benefit nor reason to create a `UriTemplate` at any point in this class.

Reply:
 I was dumb 😔. I will fix it